modules/smtp/strong_password/__init__.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import time
+import os
+import json
+import binascii
+from core.compatible import generate_token
+from database.connector import insert_to_credential_events_collection
+from database.datatypes import CredentialEvent
+
+
+class ModuleProcessor:
+    """
+    this is the processor to run after docker machine is up to grab the log files or do other needed process...
+    """
+
+    def __init__(self):
+        self.kill_flag = False
+        self.log_filename = 'tmp/ohp_smtp_honeypot_logs.txt'
+        self.log_filename_dump = 'tmp/ohp_smtp_honeypot_logs.json'
+
+    def processor(self):
+        """
+        processor function will be called as a new thread and will be die when kill_flag is True
+        :return:
+        """
+        if os.path.exists(self.log_filename):
+            os.remove(self.log_filename)  # remove if exist from past
+        while not self.kill_flag:
+            if os.path.exists(self.log_filename):
+                os.rename(self.log_filename, self.log_filename_dump)
+                data_dump = open(self.log_filename_dump).readlines()
+                for data in data_dump:
+                    data = json.loads(data)
+                    authorization = data['authorization']
+                    try:
+                        authorization = binascii.a2b_base64(authorization).\
+                                        decode('utf-8')
+                        username = authorization.split(" ")[0]
+                        password = authorization.split(" ")[1]
+                    except Exception as e:
```

Comment:
@Ali-Razmjoo : what should be done if hacker gives a wrong base64 encryption? for now i've kept it empty if not able to decode

Reply:
I think it’s best to use smtpd library in python instead